airbyte-webapp/src/pages/SettingsPage/pages/ConnectorsPage/components/CreateConnectorModal.tsx

```diff
@@ -118,8 +118,8 @@ const CreateConnectorModal: React.FC<IProps> = ({ onClose, onSubmit, errorMessag
           validateOnBlur
           validateOnChange
           validationSchema={validationSchema}
-          onSubmit={(values, { setSubmitting }) => {
-            onSubmit(values);
+          onSubmit={async (values, { setSubmitting }) => {
+            await onSubmit(values);
             setSubmitting(false);
           }}
```

Comment:
This `onSubmit` was resolving synchronously. Now it correctly `await`s the API response and the user sees a loading state:

https://user-images.githubusercontent.com/7550957/200861626-353ad933-9fbc-49ac-a39f-f09779e34de7.mov